tutorials/math/combinatorics.md

```diff
@@ -1,10 +1,215 @@
+
+Hence all we need to do is find bits which are set atleast once (by computing OR) and then multiply the final answer with $2^{n - 1}$.
+Time Complexity of the program is $O(n)$ with $O(1)$ Space Complexity.
+
+<Tabs>
+<TabItem value="cpp" label="C++">
+
+```cpp
+class Solution {
+public:
+    int subsetXORSum(vector<int>& nums) {
+        int arrayOR = 0;
+        // Do OR of whole array to obtain bits which are set atleast once
+        for(int num:nums) arrayOR |= num;
+        // Compute the final answer using the formula discussed
+        return arrayOR*(1<<(nums.size() - 1));
+    }
+};
+```
+
+</TabItem>
+</Tabs>
+
+### Example #3 [62 - Unique Paths](https://leetcode.com/problems/unique-paths/)
+
+Here our robot always goes either down or left. We know that we have to go down $m - 1$ times and go left $n - 1$ times. Thus we need to find the number of ways to arrange these. One way to visualize this is if we have $m + n - 2$ blank spaces, and we have to fill $n - 1$ of them using $L$ and remaining using $D$. Then we can just choose the number of spaces to fill with $L$ from total number of spaces. The the final solution is simply $m + n - 2 \choose n - 1$.
+
+To implement the above, use the template provided above, along with the following code:
+
+<Tabs>
+<TabItem value="cpp" label="C++">
+
+```cpp
+// Uppper limit for m + n - 2 is 198
+comb c = comb(200);
+return c.nCr(m + n - 2, n - 1);
+```
+
+</TabItem>
+</Tabs>
+
+You can check the complete solution for this problem [here](../../solutions/0000-0099/unique-paths-medium)
+
+### Example #4 [2400 - Number of Ways to Reach a Position After Exactly k Steps](https://leetcode.com/problems/number-of-ways-to-reach-a-position-after-exactly-k-steps/)
+
+Let's represent going left as $-1$ and going right as $+1$. Thus, following the same idea as before, we have $k$ blanks to fill with $+1$ and $-1$ such that there sum is equal to $startPos - endPos$.
+
+Here we can immediately see that such will be impossible in only 2 cases:
+- The parity of $k$ and $startPos - endPos$ is different.
+- The magnitude of $k$ is less than magnitude of $startPos - endPos$.
+
+After checking for above 2 cases, we know for sure that there exists a solution. Now we can just find the number of $1's$ and $-1's$ required to sum to $startPos - endPos$. Let them be $a$ and $b$ respectively. Then the solution is $startPos - endPos \choose a$ $=$ $startPos - endPos \choose b$
+
+If we assume that $endPos >= startPos$, then number of $-1$ is $(k - (endPos - startPos))/2$. If $endPos < startPos$, then we can count the number of $+1$, which will be the same. 
+
+Thus, we need to find $k \choose \frac{n - endPos + startPos}{2}$.
```

Comment:
the presentation is not nice.

![image](https://user-images.githubusercontent.com/35857179/195355406-0c58610e-b8d2-453f-a951-1fb81e01917b.png)

Reply:
Yeah, I did notice that, do you have some ideas on how to improve that, I was thinking about replacing it with another variable to make it more readable

Reply:
@wingkwong can you provide your input on this, will fix the rest of the issues.

Reply:
maybe try $C(k, \frac{endPos + startPos}{2})$ ?

````
$C(k, \frac{endPos + startPos}{2})$
````